superduperdb/backends/ibis/query.py

```diff
@@ -777,6 +843,19 @@ class IbisQueryComponent(QueryComponent):
 
     __doc__ = __doc__ + QueryComponent.__doc__  # type: ignore[operator]
 
+    @property
+    def primary_id(self):
+        assert self.type == QueryType.QUERY, 'can\'t get primary id of an attribute'
+        primary_id = []
+        for a in self.args:
```

Comment:
we should retrieve self.kwargs as well here?
since its been done above in __call__

also in __call__ we should use self.primary_id 
since this logic is redundant there.